include/sys/spa_impl.h

```diff
@@ -270,6 +270,7 @@ struct spa {
 	uint64_t	spa_scan_pass_scrub_spent_paused; /* total paused */
 	uint64_t	spa_scan_pass_exam;	/* examined bytes per pass */
 	uint64_t	spa_scan_pass_issued;	/* issued bytes per pass */
+	boolean_t	spa_resilver_defered;	/* resilver has been defered */
```

Comment:
Could you elaborate on the comment, e.g. `we are in the middle of a resilver, and another resilver is needed once this one completes.  This is set iff any vdev_resilver_deferred is set.`

That said, I'm not entirely sure that spa_resilver_deferred is needed.  Could we check if the feature is active instead?

Reply:
I actually spent a good bit of time thinking about this. The problem is that we don't always have a tx handy when we want to defer a resilver. This is the reason we increment the feature count in `dsl_scan_sync()`.